pkg/metrics/metrics.go

```diff
@@ -185,6 +184,37 @@ func (pc *PrometheusNodeConfig) Configure() error {
 	return nil
 }
 
+// retrieveMetricsEndpoint returns an existing Endpoint object or creates a new one in the operator namespace
+// if it does not already exist.
+func (pc *PrometheusNodeConfig) retrieveMetricsEndpoint() (*v1.Endpoints, error) {
```

Comment:
This functions is combining getting the endpoints or creating them. They should be separated out. Moreover, please explain why you need to create this here when it was not the case before.